src/main/java/uk/gov/pay/directdebit/common/model/subtype/CreditorId.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.pay.directdebit.common.model.subtype;
+
+import java.util.Objects;
+
+public class CreditorId {
+
+    private final String creditorId;
+
+    private CreditorId(String creditorId) {
+        this.creditorId = Objects.requireNonNull(creditorId);
+    }
+
+    public static CreditorId of(String mandateExternalId) {
+        return new CreditorId(mandateExternalId);
+    }
+
+    public static CreditorId valueOf(String creditorId) {
+        return CreditorId.of(creditorId);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CreditorId that = (CreditorId) o;
+        return Objects.equals(creditorId, that.creditorId);
```

Comment:
I think we are using the intellij default for equal and hashcode everywhere else. I am fine using the `Objects` one, though, it is quite compact.

Reply:
Changed it to `IntelliJ IDEA` style for consistency